packages/babel-plugin-proposal-decorators/src/index.js

```diff
@@ -26,75 +26,18 @@ const buildGetObjectInitializer = template(`
     })
 `);
 
-const buildInitializerWarningHelper = template(`
-    function NAME(descriptor, context){
-        throw new Error(
-          'Decorating class property failed. Please ensure that ' +
-          'proposal-class-properties is enabled and set to use loose mode. ' +
-          'To use proposal-class-properties in spec mode with decorators, wait for ' +
-          'the next major version of decorators in stage 2.'
-        );
-    }
-`);
-
-const buildInitializerDefineProperty = template(`
-    function NAME(target, property, descriptor, context){
-        if (!descriptor) return;
-
-        Object.defineProperty(target, property, {
-            enumerable: descriptor.enumerable,
-            configurable: descriptor.configurable,
-            writable: descriptor.writable,
-            value: descriptor.initializer ? descriptor.initializer.call(context) : void 0,
-        });
-    }
-`);
-
-const buildApplyDecoratedDescriptor = template(`
-    function NAME(target, property, decorators, descriptor, context){
-        var desc = {};
-        Object['ke' + 'ys'](descriptor).forEach(function(key){
-            desc[key] = descriptor[key];
-        });
-        desc.enumerable = !!desc.enumerable;
-        desc.configurable = !!desc.configurable;
-        if ('value' in desc || desc.initializer){
-            desc.writable = true;
-        }
-
-        desc = decorators.slice().reverse().reduce(function(desc, decorator){
-            return decorator(target, property, desc) || desc;
-        }, desc);
-
-        if (context && desc.initializer !== void 0){
-            desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
-            desc.initializer = undefined;
-        }
-
-        if (desc.initializer === void 0){
-            // This is a hack to avoid this being processed by 'transform-runtime'.
-            // See issue #9.
-            Object['define' + 'Property'](target, property, desc);
-            desc = null;
-        }
-
-        return desc;
-    }
-`);
-
 export default function() {
   /**
    * Add a helper to take an initial descriptor, apply some decorators to it, and optionally
    * define the property.
    */
+
   function ensureApplyDecoratedDescriptorHelper(path, state) {
     if (!state.applyDecoratedDescriptor) {
       state.applyDecoratedDescriptor = path.scope.generateUidIdentifier(
         "applyDecoratedDescriptor",
       );
-      const helper = buildApplyDecoratedDescriptor({
-        NAME: state.applyDecoratedDescriptor,
-      });
+      const helper = state.addHelper("buildApplyDecoratedDescriptor");
       path.scope.getProgramParent().path.unshiftContainer("body", helper);
```

Comment:
The problem is that these lines:
```js
const helper = buildApplyDecoratedDescriptor({
  NAME: state.applyDecoratedDescriptor,
});
path.scope.getProgramParent().path.unshiftContainer("body", helper);
```
added the helper function to the file.

On the other hand, `state.addHelper("buildApplyDecoratedDescriptor");` returns the id of the function (not the function itself), and the helper is added automatically.

So you should just remove the `ensure*` functions (which are the equivalent of `state.addHelper`), and replace calls to them (like https://github.com/babel/babel/blob/f5ef928586f592aa2d8bb60120ae58833d17e0db/packages/babel-plugin-proposal-decorators/src/index.js#L273) with `addHelper(" ... ")`.